### PR TITLE
resourcemanger: add more test about assign tokens and comments

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: Ref #4399 

### What is changed and how does it work?

```commit-message
resourcemanger: add more test about assign tokens and comments
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by nolouch** [comment]
cc @glorv

**Review by glorv — COMMENTED** [review]
LGTM

File: pkg/mcs/resourcemanager/server/token_buckets_test.go [hunk 1]
```diff
@@ -123,3 +124,65 @@ func TestGroupTokenBucketRequest(t *testing.T) {
...
+		expectedTrickle           int64
+	}{
+		/* requestTokens, assignedTokens, currentGlobalBucketTokens, trickleTime */
+		{50000, 50000, 0, 0},
+		{50000, 30000, -20000, defaultTrickleDuration},
+		{30000, 15000, -25000, defaultTrickleDuration},
+		{15000, 12500, -27500, defaultTrickleDuration},
+		{12500, 11250, -28750, defaultTrickleDuration},
+		{11250, 10625, -29375, defaultTrickleDuration},
+		// PR_PER_SEC is close to 2000, RU_PER_SEC =  assignedTokens / TrickleTime.
```
Comment (okJiang) [inline pkg/mcs/resourcemanager/server/token_buckets_test.go:163]:
```suggestion
		// RU_PER_SEC is close to 2000, RU_PER_SEC =  assignedTokens / TrickleTime.
```

Comment (okJiang) [reply pkg/mcs/resourcemanager/server/token_buckets_test.go:163]: What is the unit for TrickleTime?

> const defaultTrickleDuration = int64(time.Second) * 5 / int64(time.Millisecond)

Comment (nolouch) [reply pkg/mcs/resourcemanager/server/token_buckets_test.go]: default `ms`

File: pkg/mcs/resourcemanager/server/token_buckets_test.go [hunk 2]
```diff
@@ -144,45 +144,46 @@ func TestGroupTokenBucketRequestLoop(t *testing.T) {
 	gtb.Tokens = 50000
 
 	const timeIncrement = 5 * time.Second
-	const defaultTrickleDuration = int64(time.Second) * 5 / int64(time.Millisecond)
+	const targetPeriod = 5 * time.Second
+	const defaultTrickleMs = int64(targetPeriod) / int64(time.Millisecond)
```
Comment (okJiang) [inline pkg/mcs/resourcemanager/server/token_buckets_test.go:148]: The `Millisecond` here is meaningless
```suggestion
	const defaultTrickleMs = int64(targetPeriod) / int64(1000)
```

File: pkg/mcs/resourcemanager/server/token_buckets_test.go [hunk 3]
```diff
@@ -144,45 +144,46 @@ func TestGroupTokenBucketRequestLoop(t *testing.T) {
...
-		{10000.6103515625, 10000.30517578125, -29999.69482421875, defaultTrickleDuration},
-		{10000.30517578125, 10000.152587890625, -29999.847412109375, defaultTrickleDuration},
-		{10000.152587890625, 10000.0762939453125, -29999.9237060546875, defaultTrickleDuration},
-		{10000.0762939453125, 10000.038146972656, -29999.961853027343, defaultTrickleDuration},
+		{50000, 30000, -20000, defaultTrickleMs},
+		{30000, 15000, -25000, defaultTrickleMs},
+		{15000, 12500, -27500, defaultTrickleMs},
+		{12500, 11250, -28750, defaultTrickleMs},
+		{11250, 10625, -29375, defaultTrickleMs},
+		// RU_PER_SEC is close to 2000, RU_PER_SEC =  assignedTokens / TrickleTime.
```
Comment (okJiang) [inline pkg/mcs/resourcemanager/server/token_buckets_test.go:164]: Please add the unit conversion

```suggestion
		// RU_PER_SEC is close to 2000, RU_PER_SEC =  assignedTokens / (TrickleTime / 1000).
```

**Review by okJiang — APPROVED** [review]
rest lgtm

File: pkg/mcs/resourcemanager/server/token_buckets.go [hunk 4]
```diff
@@ -409,6 +409,11 @@ func (ts *TokenSlot) assignSlotTokens(requiredToken float64, targetPeriodMs uint
 	//               |
 	// grant_rate 0  ------------------------------------------------------------------------------------
 	//         loan      ***    k*period_token    (k+k-1)*period_token    ***      (k+k+1...+1)*period_token
+
+	// loadCoefficient is relative to the capacity of load RUs.
```
Comment (HuSharp) [inline pkg/mcs/resourcemanager/server/token_buckets.go:413]:
```suggestion
	// loanCoefficient is relative to the capacity of load RUs.
```

**Review by HuSharp — APPROVED** [review]

**Pull request merged**